### PR TITLE
Axis angle

### PR DESCRIPTION
**Pull request by jgoppert**
This closes https://github.com/PX4/Matrix/issues/18

@tumbili @bkueng @LorenzMeier 

to_axis_angle, from_axis_angle should now be deprecated.

### CODE REVIEW

File: matrix/AxisAngle.hpp [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+     *
+     * @param q quaternion
+     */
+    AxisAngle(const Quaternion<Type> &q) :
+        Vector<Type, 3>()
+    {
+        AxisAngle &v = *this;
+        Type angle = (Type)2.0f*acosf(q(0));
+        Type mag = sinf(angle/2.0f);
+        v(0) = angle*q(1)/mag;
```
Comment (bkueng) [inline matrix/AxisAngle.hpp]: If `angle` = 0, you have a division by 0

File: matrix/AxisAngle.hpp [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+     * Instance is initialized from a dcm representing coordinate transformation
+     * from frame 2 to frame 1.
+     *
+     * @param dcm dcm to set quaternion to
+     */
+    AxisAngle(const Dcm<Type> &dcm) :
+        Vector<Type, 3>()
+    {
+        AxisAngle &v = *this;
+        v = Quaternion<float>(dcm);
```
Comment (bkueng) [inline matrix/AxisAngle.hpp]: I think this should be `Type` instead of float

File: matrix/AxisAngle.hpp (hunk 1 above)
Comment (jgoppert) [inline matrix/AxisAngle.hpp]: Thanks for the catch.

File: matrix/AxisAngle.hpp (hunk 2 above)
Comment (jgoppert) [inline matrix/AxisAngle.hpp]: Right, will fix this as well.

**Pull request merged**